samcli/lib/build/app_builder.py

```diff
@@ -0,0 +1,137 @@
+"""
+Builds the application
+"""
+
+import os
+import shutil
+from aws_lambda_builders.builder import LambdaBuilder
+
+
+class ApplicationBuilder(object):
+    """
+    Class to build an entire application. Currently, this class builds Lambda functions only, but there is nothing that
+    is stopping this class from supporting other resource types. Building in context of Lambda functions refer to
+    converting source code into artifacts that can be run on AWS Lambda
+    """
+
+    _builder_capabilities = {
+        "python2.7": {
+            "language": "python",
+            "dependency_manager": "pip",
+            "application_framework": None,
+            "manifest_name": "requirements.txt"
+        },
+        "python3.6": {
+           "language": "python",
+           "dependency_manager": "pip",
+           "application_framework": None,
+           "manifest_name": "requirements.txt"
+        }
+    }
+
+    def __init__(self, function_provider, build_dir, source_root, use_container=False, parallel=False):
+        """
+        Initialize the class
+
+        Parameters
+        ----------
+        function_provider : samcli.commands.local.lib.sam_function_provider.SamFunctionProvider
+            Provider that can vend out functions available in the SAM template
+
+        build_dir : str
+            Path to the directory where we will be storing built artifacts
+
+        source_root : str
+            Path to a folder. Use this folder as the root to resolve relative source code paths against
+
+        use_container : bool
+            Optional. Set to True if you want to run the builds on an container that simulates AWS Lambda environment
+
+        parallel : bool
+            Optional. Set to True to build each function in parallel to improve performance
+        """
+        self.function_provider = function_provider
+        self.build_dir = build_dir
+        self.source_root = source_root
+
+        self.use_container = use_container
+        self.parallel = parallel
+
+    def build(self):
+        """
+        Build the entire application
+
+        Returns
+        -------
+        dict
+            Returns the path to where each resource was built as a map of resource's LogicalId to the path string
+        """
+
+        result = {}
+
+        for lambda_function in self.function_provider.get_all():
```

Comment:
Is there a way I can specify a specific function to build? I am thinking of a use case for fast dev cycles on one of the many functions I define.

Reply:
Not at the moment. It might be a good future option